_posts/2021-02-10-native-k8s-with-ha.md

```diff
@@ -0,0 +1,112 @@
+
+# Example: Application cluster with HA
+
+Please note that you need a running Kubernetes cluster and to get kube config properly set to follow along.
+You can use `kubectl get nodes` for verifying that you’re all set.
+In this blog post, we’re using [minikube](https://minikube.sigs.k8s.io/docs/start/) for local testing.
+
+1. Build a Docker image with the Flink job (my-flink-job.jar) baked in
+
+        FROM flink:1.12.1
+        RUN mkdir -p $FLINK_HOME/usrlib
+        COPY /path/of/my-flink-job.jar $FLINK_HOME/usrlib/my-flink-job.jar
+
+    Use the above Dockerfile to build a user image and then push it to your remote image repository.
+
+2. Start a Flink application cluster
+
+        $ ./bin/flink run-application -d -p 4 -t kubernetes-application \
+            -Dkubernetes.cluster-id=k8s-ha-app-1 \
+            -Dkubernetes.container.image=<user-image> \
+            -Dkubernetes.jobmanager.cpu=0.5 -Dkubernetes.taskmanager.cpu=0.5 \
+            -Dtaskmanager.numberOfTaskSlots=4 \
+            -Dkubernetes.rest-service.exposed.type=NodePort \
+            -Dhigh-availability=org.apache.flink.kubernetes.highavailability.KubernetesHaServicesFactory \
+            -Dhigh-availability.storageDir=s3://flink-bucket/flink-ha \
+            -Drestart-strategy=fixed-delay -Drestart-strategy.fixed-delay.attempts=10 \
+            -Dcontainerized.master.env.ENABLE_BUILT_IN_PLUGINS=flink-s3-fs-hadoop-1.12.1.jar \
+            -Dcontainerized.taskmanager.env.ENABLE_BUILT_IN_PLUGINS=flink-s3-fs-hadoop-1.12.1.jar \
+            local:///opt/flink/usrlib/my-flink-job.jar
+
+3. Access the Flink Web UI (http://minikube-ip-address:node-port) and check that the job is running
+
+        2021-02-05 17:26:13,403 INFO  org.apache.flink.kubernetes.KubernetesClusterDescriptor      [] - Create flink application cluster k8s-ha-app-1 successfully, JobManager Web Interface: http://192.168.64.21:32388
+
+    You could find a similar log in the Flink client and get the JobManager web interface URL.
+
+4. Kill the JobManager to simulate failure
+
+        kubectl exec {jobmanager_pod_name} -- /bin/sh -c "kill 1"
+
+5. Verify that the job recovers from the latest successful checkpoint
+    
+    Refresh the web dashboard until the new JobManager is launched, and then search for the following JobManager logs to verify that the job recovers from the latest successful checkpoint:
+
+        2021-02-05 09:44:01,636 INFO  org.apache.flink.runtime.checkpoint.CheckpointCoordinator    [] - Restoring job 00000000000000000000000000000000 from Checkpoint 101 @ 1612518074802 for 00000000000000000000000000000000 located at <checkpoint-not-externally-addressable>.
+
+6. Cancel the job
+
+    The job can be cancelled in the Flink the WebUI, or using the following command:
+
+        ./bin/flink cancel -t kubernetes-application -Dkubernetes.cluster-id=<ClusterID> <JobID>
```

Comment:
```suggestion
        ./bin/flink cancel --target kubernetes-application -Dkubernetes.cluster-id=<ClusterID> <JobID>
```
nit: Just for consistency purposes in case we want to go with the long parameter approach.